sklearn/decomposition/dict_learning.py

```diff
@@ -875,7 +875,7 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
         return dictionary.T
 
 
-class SparseCodingMixin(TransformerMixin):
+class SparseCodingMixin(ComponentsMixin, TransformerMixin):
```

Comment:
So we've silently started to add mixins on the "right" side? :P I did need to do the same in for feature names as well.

Reply:
The right side of what? Honestly I would rather start using base-classes. Here there's two mixins. But we do need to have the more specialized on the left, otherwise it doesn't work.

Reply:
by "right" I meant "correct".

Reply:
lol so much for me being able to tell left from right.